platformio/commands/device/filters/plotter.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2014-present PlatformIO <contact@platformio.org>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# This requires the ~/platformio/package/tool-serialplotter package
+#
+import subprocess
+import socket
+import os
+from platformio.commands.device import DeviceMonitorFilter
+
+PORT = 19200
+
+class SerialPlotter(DeviceMonitorFilter):
+    NAME = "serial_plotter"
+
+    def __init__(self, *args, **kwargs):
+        super(SerialPlotter, self).__init__(*args, **kwargs)
+        self.buffer = ''
+        self.plotter_py = os.path.expanduser('~') + '/.platformio/packages/tool-serialplotter/serialPlotter.py'
```

Comment:
Will this work for Windows users? Better not use `os.path.join()` to concatenate paths?

Reply:
Hi @maxgerhardt ,
Thank you for the suggestion. I'll modify the code on my branch for this.

Reply:
Oh, in fact it can't be assumed that PlatformIO is even installed in that path since it can be arbitrary. You should use `get_project_core_dir()` per https://github.com/platformio/platformio-core/blob/311e10f91e825348df97c0701f575f2b42260b57/platformio/debug/config/base.py#L211

Reply:
Great!  Actually that's what I really needed. Thanks.

Reply:
By the way, since I started changing the serialPlotter tool itself after getting @ivankravets's advice, I may change some part of this filter to take the tools modification. And I may work on it this Weekend. So let's continue this thread early next week. Thanks.

Reply:
Hi @ivankravets and @maxgerhardt ,

I reworked the serialplotter python command and it’s renamed to arduplot taking @ivankravets idea of naming.
Github repo → https://github.com/yhur/arduplot

1. And I pushed it to pypi.org, so it can be installed with pip.
2. this serial_plotter filter is also updated.

Would you take a look and give me your feedback before the merge?